Documentation/teaching/labs/interrupts.rst

```diff
@@ -968,44 +990,72 @@ character.
 	  Where scancode is the value of the data register, and ch is
 	  the value returned by the get_ascii() function.
 
-
-4. Store characters to the buffer
----------------------------------
+Store characters to the buffer
+...............................
 
 We want to collect the pressed characters (not the other keys) into
-circular a buffer that can be consumed from user space. For this step
-follow the sections marked with **TODO 4** in the skeleton.
-
-Implement :c:func:`get_char` in a similar way to :c:func:`put_char`.
+circular a buffer that can be consumed from user space.
 
 Update the interrupt handler to add a pressed ASCII character to the
 end of the device buffer. If the buffer is full, the character will be
 discarded.
 
-.. hint:: To get the device data from the interrupt handler use the
-	  following construct:
+.. hint:: The device buffer is the field :c:type:`buf` in the device's
+    :c:type:`struct kbd`. To get the device data from the interrupt handler
+    use the following construct:
 
 	  .. code-block:: c
 
 	     struct kbd *data = (struct kbd *) dev_id;
 
+    The buffer's dimension is located in :c:type:`struct kbd`'s field,
+    :c:type:`count`. The :c:type:`put_idx` and :c:type:`get_idx` fields
+    specify the next writing and reading index. Take a look at the
+    :c:func:`put_char` function's implementation to observe how the data is
+    added to the circular buffer.
+
 .. attention:: Synchronize the access to the buffer and the helper
 	       indexes with a spinlock.
```

Comment:
This is not indented correctly and doesn't appear correctly on the page

Reply:
Fixed.